websites/T/The Church of Jesus Christ of Latter-day Saints/presence.ts

```diff
@@ -0,0 +1,739 @@
+
+  // Fetch the song style
+  let style = parsePlayerStyle()
+  if (style === undefined) {
+    style = undefined // we don't need to set the style if it's not available
+  }
+  const previousStyle: string | undefined = listeningStyle
+  listeningStyle = style
+
+  // Attempt to parse the start and end timestamps
+  const startTimestamp: number | undefined = parsePlayerTimestamp1()
+  const endTimestamp: number | undefined = parsePlayerTimestamp2()
+  if (startTimestamp !== undefined && endTimestamp !== undefined) {
+    const timestamps = getTimestamps(startTimestamp, endTimestamp)
+    presenceData.startTimestamp = timestamps[0]
+    presenceData.endTimestamp = timestamps[1]
+  }
+  else {
+    presenceData.startTimestamp = currentTimestamp
+    delete presenceData.endTimestamp
+  }
+
+  // Attempt to parse if the player is playing
+  const isPlaying: boolean = parsePlayerIsPlaying()
+  if (isPlaying) {
+    presenceData.smallImageKey = Assets.Play
+    presenceData.smallImageText = 'Playing'
+  }
+  else {
+    presenceData.smallImageKey = Assets.Pause
+    presenceData.smallImageText = 'Paused'
+  }
+
+  // If we are already listening, update the timestamps and return
+  if (currentState === 'listening' && listeningLocation === location
+    && listeningTitle === previousTitle && listeningStyle === previousStyle
+    && listeningAlbum === previousAlbum) {
+    return
+  }
+  currentState = 'listening'
+  listeningLocation = location
+
+  // Attempt to parse the image thumbnail
+  let thumbnail: string | undefined = parsePlayerThumbnail()
+  if (thumbnail === undefined) {
+    thumbnail = ActivityAssets.Logo // Fallback to the logo if no thumbnail is found
+  }
+
+  // Create a short title if the title is too long
+  let shortTitle: string | undefined
+  if (listeningTitle.length > 15) {
+    shortTitle = `${listeningTitle.substring(0, 15)}...`
+  }
+  else {
+    shortTitle = listeningTitle
+  }
+
+  // Construct the state
+  let stateText: string | undefined
+  if (listeningAlbum !== undefined && listeningStyle !== undefined) {
+    stateText = `${listeningAlbum} - ${listeningStyle}`
+  }
+  else if (listeningAlbum !== undefined) {
+    stateText = listeningAlbum
+  }
+  else if (listeningStyle !== undefined) {
+    stateText = listeningStyle
+  }
+  else {
+    stateText = undefined // No state text if no album or style is available
+  }
+
+  presenceData = {
+    type: ActivityType.Listening,
+    largeImageKey: thumbnail,
+    smallImageKey: Assets.Pause,
+    smallImageText: 'Paused',
+    startTimestamp: currentTimestamp,
+    details: listeningTitle,
+    state: stateText,
+    buttons: [
+      {
+        label: `Listen to ${shortTitle}`,
+        url: document.location.href,
+      },
+    ],
+  }
+
+  // Perform triple-update
+  tripleUpdate()
+}
+
+function parseScriptureReference(): string | undefined {
+  // Attempt to parse via content title
+  const contentTitle = document.querySelector('.contentTitle-JbPZw')
```

Comment:
Try using a selector without the generated class names. For this example, you can try something like this:
```ts
const contentTitle = document.querySelector('[class*=contentTitle]')
```